### PR TITLE
📖 ClusterClass proposal: remove builtin labels & annotations variables, add new

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
For now we won't implement builtin labels/annotations variables. So let's update the ClusterClass proposal to reflect that to avoid confusing users.

Depending on the outcome of the discussion https://github.com/kubernetes-sigs/cluster-api/issues/5577 we can potentially introduce the labels/annotations variables later and add them to the proposal again.

This PR also syncs the other labels with the latest results from #5534.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/proposals/202105256-cluster-class-and-managed-topologies.md [hunk 1]
```diff
@@ -571,11 +571,11 @@ Note: Builtin variables are defined in [Builtin variables](#builtin-variables) b
 #### Builtin variables
 
 It’s also possible to use so-called builtin variables in addition to user-defined variables. The following builtin variables are available:
-- `cluster.{name,namespace,labels,annotations}`
+- `cluster.{name,namespace}`
 - `cluster.topology.{version,class}`
-- `cluster.topology.controlPlane.{labels,annotations,replicas,version}`
+- `cluster.topology.controlPlane.{replicas,version}`
```
Comment (fabriziopandini) [inline docs/proposals/202105256-cluster-class-and-managed-topologies.md:576]: Am I wrong or the variable names in the current PR has been simplified, e.g.
```suggestion
- `controlPlane.{replicas,version}`
```
Same below

Comment (sbueringer) [reply docs/proposals/202105256-cluster-class-and-managed-topologies.md:576]: Yup. I didn't make those changes as they are already in the other PR. But I'll sync that.

Comment (sbueringer) [reply docs/proposals/202105256-cluster-class-and-managed-topologies.md]: Done
This PR now contains the latest version including changes from the patch engine PR.

**Comment by fabriziopandini** [comment]
/lgtm
Also commented on #5577

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Review by fabriziopandini — COMMENTED** [review]
/approve

**Pull request merged**